src/Client/MultiplexedConnections.cpp

```diff
@@ -319,24 +319,21 @@ Packet MultiplexedConnections::receivePacketUnlocked(AsyncCallback async_callbac
         throw Exception(ErrorCodes::NO_AVAILABLE_REPLICA, "Logical error: no available replica");
 
     Packet packet;
+    try
     {
         AsyncCallbackSetter async_setter(current_connection, std::move(async_callback));
```

Comment:
Previously this `AsyncCallbackSetter` could be destroyed after `invalidateReplica()` is called, so we could touch connection that was already returned to connection pool